patches/vtk5.10.1VS2015.patch

```diff
@@ -1,5 +1,62 @@
+From 5827f3758e67fc73f293d047d8c65593d96091d9 Mon Sep 17 00:00:00 2001
+From: Mathilde Merle <mathilde.merle@ihu-liryc.fr>
+Date: Wed, 24 Apr 2019 14:52:49 +0200
+Subject: [PATCH] vtk5.10.1VS2015
+
+---
+ CMake/vtkCompilerExtras.cmake                 |   12 +-
+ Charts/vtkControlPointsItem.cxx               |    1 +
+ Charts/vtkPiecewisePointHandleItem.cxx        |    2 +
+ Common/vtkOStreamWrapper.cxx                  |    5 +-
+ Hybrid/vtk3DSImporter.cxx                     |    8 +-
+ Hybrid/vtkVectorText.cxx                      | 1512 ++++++++---------
+ IO/vtkEnSightGoldBinaryReader.cxx             |   22 +-
+ IO/vtkEnSightReader.cxx                       |   10 +-
+ IO/vtkXMLUnstructuredDataReader.cxx           |    4 +-
+ Infovis/vtkAdjacencyMatrixToEdgeTable.cxx     |    1 +
+ Infovis/vtkConvexHull2D.cxx                   |    2 +
+ Infovis/vtkNormalizeMatrixVectors.cxx         |    1 +
+ Infovis/vtkPairwiseExtractHistogram2D.cxx     |    1 +
+ Rendering/vtkInteractorStyleImage.cxx         |   11 +-
+ Rendering/vtkXOpenGLRenderWindow.cxx          |    2 +-
+ Utilities/vtkhdf5/ConfigureChecks.cmake       |    2 +
+ Utilities/vtkhdf5/Resources/HDF5Tests.c       |    2 +-
+ Utilities/vtklibxml2/config_cmake.h.in        |    4 +-
+ Utilities/vtktiff/CMakeLists.txt              |    1 +
+ Utilities/vtktiff/tif_config.h.in             |   12 +-
+ .../vtkParallelCoordinatesRepresentation.cxx  |    1 +
+ 21 files changed, 812 insertions(+), 804 deletions(-)
+
+diff --git a/CMake/vtkCompilerExtras.cmake b/CMake/vtkCompilerExtras.cmake
+index 6811ac2597..4582f28763 100644
+--- a/CMake/vtkCompilerExtras.cmake
++++ b/CMake/vtkCompilerExtras.cmake
+@@ -22,22 +22,12 @@ if(CMAKE_COMPILER_IS_GNUCXX)
+     #  "-Wl,--fatal-warnings -Wl,--no-undefined -lc ${CMAKE_SHARED_LINKER_FLAGS}")
+   endif()
+ 
+-  # Now check if we can use visibility to selectively export symbols
+-  exec_program(${CMAKE_C_COMPILER} ARGS --version OUTPUT_VARIABLE
+-    _gcc_version_info)
+-  string (REGEX MATCH "[345]\\.[0-9]\\.[0-9]" 
+-    _gcc_version "${_gcc_version_info}")
+-  if(NOT _gcc_version)
+-    string (REGEX REPLACE ".*\\(GCC\\).* ([34]\\.[0-9]) .*" "\\1.0" 
+-      _gcc_version "${_gcc_version_info}")
+-  endif()
+-  
+   # GCC visibility support, on by default and in testing.
+   check_cxx_compiler_flag(-fvisibility=hidden HAVE_GCC_VISIBILITY)
+   option(VTK_USE_GCC_VISIBILITY "Use GCC visibility support if available." ON)
+   mark_as_advanced(VTK_USE_GCC_VISIBILITY)
+ 
+-  if(${_gcc_version} VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
++  if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.2.0 AND BUILD_SHARED_LIBS
```

Comment:
we can use directly CMAKE_CXX_COMPILER_VERSION, it avoids problem with gcc 6 and above